duecredit/collector.py

```diff
@@ -259,8 +258,21 @@ def cite(self, entry, **kwargs):
                 version = external_versions[package]
             citation.version = version
 
+        self.citations[CitationKey(path, entry_key)] = citation
```

Comment:
better make  
```python
@property
def key(self):
  return CitationKey(path, entry_key)
```
and use it instead of explicit CitationKey here, so if later we decide to e.g. add version -- it would be right in the Citation code